Makefile.core.mk

```diff
@@ -119,33 +123,29 @@ gen:
 gen-check:
 	@scripts/gen-testdata.sh -c
 
-test: V8_BUILD_CONFIG=$(BAZEL_CONFIG_DEV)
 test: prerun_v8_build
 	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_DEV) //src/envoy:envoy
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_DEV) -- $(BAZEL_TEST_TARGETS)
-	env ENVOY_PATH=$(BAZEL_ENVOY_PATH) GO111MODULE=on go test -timeout 30m ./...
+	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT) $(TEST_ENVOY_TARGET)
+	if [ -n "$(BAZEL_TEST_TARGETS)" ]; then \
+	  export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT) $(BAZEL_TEST_ARGS) -- $(BAZEL_TEST_TARGETS); \
+	fi
+	if [ -n "$(E2E_TEST_TARGETS)" ]; then \
+	  env ENVOY_PATH=$(TEST_ENVOY_PATH) GO111MODULE=on go test -timeout 30m $(E2E_TEST_TARGETS); \
+	fi
 
-test_asan: V8_BUILD_CONFIG=$(BAZEL_CONFIG_ASAN)
-test_asan: prerun_v8_build
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_ASAN) //src/envoy:envoy
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_ASAN) -- $(BAZEL_TEST_TARGETS)
-	env ENVOY_PATH=$(BAZEL_ENVOY_PATH) ASAN=true GO111MODULE=on go test -timeout 30m ./...
+test_asan: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_ASAN)
+test_asan: test
 
-test_tsan: V8_BUILD_CONFIG=$(BAZEL_CONFIG_TSAN)
-test_tsan: prerun_v8_build
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && \
-	bazel $(BAZEL_STARTUP_ARGS) build $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_TSAN) //src/envoy:envoy
-	export PATH=$(PATH) CC=$(CC) CXX=$(CXX) && bazel $(BAZEL_STARTUP_ARGS) test $(BAZEL_BUILD_ARGS) $(BAZEL_CONFIG_TSAN) -- $(BAZEL_TEST_TARGETS)
-	env ENVOY_PATH=$(BAZEL_ENVOY_PATH) TSAN=true GO111MODULE=on go test -timeout 30m ./...
+test_tsan: BAZEL_CONFIG_CURRENT = $(BAZEL_CONFIG_ASAN)
+test_tsan: test
+
+test_centos: BAZEL_BUILD_ARGS := $(CENTOS_BUILD_ARGS) $(BAZEL_CONFIG_CURRENT)
```

Comment:
Should concat with `BAZEL_BUILD_ARGS` instead of `BAZEL_CONFIG_CURRENT `?

Reply:
Thanks for the catch. I've fixed it.